tests/test_basic_regression.py

```diff
@@ -152,3 +152,66 @@ def test_regression(pdb, options, tmp_path):
     run_propka(options, pdb_path, tmp_path)
     if ref_path is not None:
         compare_output(pdb, tmp_path, ref_path)
+
+
+def run_propka_stream(options, input_file, tmp_path):
```

Comment:
Does it also work with StringIO buffers?

Reply:
Yes :) I've moved everything over to a separate `test_streamio.py` file (and added an __init__.py), it was getting a tad bit crowded for a file that was really meant to test basic regression. I've also:
a) tests both TextIO and StringIO inputs.
b) reduced the number of test cases (all that needed to be checked was if it gave the same results and also allowed arguments).
c) switched the usage of tmp_dir over to pytest's tmpdir, which seems a lot cleaner to use.
d) added a test to capture the TypeError that should be thrown if you don't pass a value to `filename` in `read_molecule_file`.